File: Projects/SchoolAppApp/SchoolAppointmentApp BackUpV1/Entities/Product.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace SchoolAppointmentApp.Entities;

public class Product
{
    public int ProductId { get; set; }
    public string ProductName { get; set; } = null!;
    public required string ProductImageRoot { get; set; }
    public required string Description { get; set; }
    [Range(0, int.MaxValue)] public required int PointCost { get; set; }
    [Range(0, int.MaxValue)] public required int AvailableQuantity { get; set; }
}

File: Projects/SchoolAppApp/types/Product.ts
export type ProductInitialType = {
  products: ShoppingReturn[];
  targetProducts?: ShoppingReturn;
};

export type ShoppingReturn = {
  productId: number;
  productImageRoot: string;
  productName: string;
  description: string;
  pointCost: number;
  availableQuantity: number;
};

export type LoadProductsPayload = {
  SearchString?: string;
  RequestExpandTimes: number;
};



